pkg/prometheus/server/operator.go

```diff
@@ -964,6 +972,44 @@ func (c *Operator) sync(ctx context.Context, key string) error {
 	return nil
 }
 
+// add or remove finalizers for the Prometheus resource.
+func (c *Operator) syncFinalizers(ctx context.Context, p *monitoringv1.Prometheus, key string) (bool, error) {
+	if c.configResourcesStatusEnabled {
```

Comment:
(suggestion) a good practice is to check first for the exit condition(s).
```suggestion
	if !c.configResourcesStatusEnabled {
		return false, nil
	}
```